pkg/cache/cache.go

```diff
@@ -440,16 +443,21 @@ func (c *Cache) UpdateClusterQueue(cq *kueue.ClusterQueue) error {
 	return nil
 }
 
-func (c *Cache) DeleteClusterQueue(cq *kueue.ClusterQueue) {
+func (c *Cache) DeleteClusterQueue(ctx context.Context, cq *kueue.ClusterQueue) {
 	c.Lock()
 	defer c.Unlock()
 	_, ok := c.hm.ClusterQueues[cq.Name]
 	if !ok {
 		return
 	}
-	if features.Enabled(features.LocalQueueMetrics) {
```

Comment:
I think you still want the feature gate here as long as this feature is alpha/beta.

Also, you should make sure to bump the feature gate to beta if you are intending to bump this to beta.

Reply:
I think you could use ShouldReportLocalQueueMetrisc here. YOu check feature gate in that code and you also check enabled.